sound/soc/sof/control.c

```diff
@@ -11,8 +11,41 @@
 /* Mixer Controls */
 
 #include <linux/pm_runtime.h>
+#include <linux/leds.h>
 #include "sof-priv.h"
 
+static void update_acpi_mute_led(struct snd_sof_control *scontrol,
+				 struct snd_kcontrol *kcontrol,
+				 struct snd_ctl_elem_value *ucontrol)
+{
+	struct snd_sof_dev *sdev = scontrol->sdev;
+	unsigned int mask;
+	unsigned int val;
+
+	mask = 1U << snd_ctl_get_ioffidx(kcontrol, &ucontrol->id);
+
+	if (ucontrol->value.integer.value[0] ||
+	    ucontrol->value.integer.value[1])
+		scontrol->led_ctl.led |= mask;
+	else
+		scontrol->led_ctl.led &= ~mask;
+
+	val = !scontrol->led_ctl.led;
```

Comment:
is this a switch on-> mute off conversion?

Reply:
this part was also from hda... so: 
if ucontrol->value.integer.value[0] is non-zero
my value after bit twiddling is:
1 for "switch on" what is "mute off" what should convert to "led off"
0 for "switch off" what is "mute on" what should convert to "led on"
So I can just remove the ! and use the value directly. I don't know if would be somehow clearer to inverse the value.